src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1601,29 +1624,16 @@ default boolean isFromGuild()
      * <p>Messages store reactions by keeping a list of reaction names.
      *
      * <p>An instance of the related {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} can be
-     * obtained through this method by using the emoji's id.
-     *
-     *
-     * <p>The following {@link net.dv8tion.jda.api.requests.ErrorResponse ErrorResponses} are possible:
-     * <ul>
-     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
-     *     <br>The request was attempted after the account lost access to the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}
-     *         due to {@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ} being revoked
-     *     <br>Also can happen if the account lost the {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}</li>
-     *
-     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_MESSAGE UNKNOWN_MESSAGE}
-     *         The reaction request was attempted after the Message had been deleted.</li>
-     * </ul>
+     * obtained through this method by using the emote's id.
      *
      * @param  id
-     *         The string id of the reaction emoji.
+     *         The string id of the reaction emote.
      *
      * @throws java.lang.UnsupportedOperationException
      *         If this is not a Received Message from {@link net.dv8tion.jda.api.entities.MessageType#DEFAULT MessageType.DEFAULT}
-     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
-     *         If the MessageChannel this message was sent in was a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the
-     *         logged in account does not have {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY} in the channel.
      * @throws java.lang.IllegalArgumentException
+     *         If the provided id is null.
+     * @throws java.lang.NumberFormatException
```

Comment:
This isn't true. `Checks.isSnowflake` throws `IllegalArgumentException`
Additionally, an id being `null` already makes it "not a snowflake", thus illegal.

This should be reverted.

Reply:
In this case, `MiscUtil.parseSnowflake` was called without an `isSnowflake` check. I added the check and reverted the documentation change.